lib/ber/reader.js

```diff
@@ -225,25 +226,47 @@ Reader.prototype._readTag = function(tag) {
   if (o === null)
     return null;
 
-  if (this.length > 4)
+  if (this.length > 8)
     throw newInvalidAsn1Error('Integer too long: ' + this.length);
 
   if (this.length > this._size - o)
     return null;
   this._offset = o;
 
-  var fb = this._buf[this._offset];
-  var value = 0;
-
-  for (var i = 0; i < this.length; i++) {
-    value <<= 8;
-    value |= (this._buf[this._offset++] & 0xff);
-  }
+  if (!this.length)
+    return -1;
```

Comment:
The original semantics were the same. The only difference here is returning early to avoid creating a 0 length buffer instance.